src/NetscriptFunctions/Corporation.ts

```diff
@@ -677,13 +678,19 @@ export function NetscriptCorporation(): InternalAPI<NSCorporation> {
     ...warehouseAPI,
     ...officeAPI,
     hasCorporation: () => () => !!Player.corporation,
-    getConstants: (ctx) => () => {
-      checkAccess(ctx);
+    getConstants: (ctx) => {
       /* TODO 2.2: possibly just rework the whole corp constants structure to be more readable, and just use cloneDeep
        *           to provide it directly to player.
        * TODO 2.2: Roll product information into industriesData, there's no reason to look up a product separately */
-      return cloneDeep(omit(corpConstants, "fundingRoundShares", "fundingRoundMultiplier", "valuationLength"));
       // TODO: add functions for getting materialInfo and research info
+      const constants = cloneDeep(
+        omit(corpConstants, "fundingRoundShares", "fundingRoundMultiplier", "valuationLength"),
+      );
+
+      return () => {
+        checkAccess(ctx);
+        return constants;
+      };
     },
```

Comment:
If the player modifies that 1-time clone (e.g. they accidentally did an assignment instead of comparison in an if statement conditional), that alteration will persist every time they call getConstants which is not desirable.

Reply:
maybe we should punish players for this? xD

Reply:
sorry, but I am a bit lost here - sometimes we say something like "let player try to hack the game and do weird stuff" but sometimes we try to protect them from something which does not even have impact on game (since this is a cloned value)

Reply:
It makes it so the game will never provide them with correct constants through getConstants unless they reload the page. We want our API to continue to provide the correct data to the player, no matter what they do with that data once they have it.

Reply:
@d0sboots, what do you think about it?
I am personally fine with any option you chose guys, you are maintainers anyway :)

Reply:
I'm with Snarling here: Changing this creates a weird failure case, and requires more code, for a speedup that won't/shouldn't ever be observed.

Reply:
👍